plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeGcsConnectorSmokeTest.java

```diff
@@ -0,0 +1,294 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.plugin.deltalake.DeltaLakeQueryRunner.DELTA_CATALOG;
+import static io.trino.plugin.deltalake.DeltaLakeQueryRunner.createDockerizedDeltaLakeQueryRunner;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+import static java.util.regex.Matcher.quoteReplacement;
+
+/**
+ * This test requires these variables to connect to GCS:
+ * - gcp-storage-bucket: The name of the bucket to store tables in. The bucket must already exist.
+ * - gcp-credentials-key: A base64 encoded copy of the JSON authentication file for the service account used to connect to GCP.
+ *   For example, `cat service-account-key.json | base64`
+ */
+public class TestDeltaLakeGcsConnectorSmokeTest
+        extends BaseDeltaLakeConnectorSmokeTest
+{
+    private static final FileAttribute<?> READ_ONLY_PERMISSIONS = PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("rw-r--r--"));
+
+    private final String gcpStorageBucket;
+    private final Path gcpCredentialsFile;
+    private final FileSystem fileSystem;
+
+    @Parameters({"gcp-storage-bucket", "gcp-credentials-key"})
+    public TestDeltaLakeGcsConnectorSmokeTest(String gcpStorageBucket, String gcpCredentialKey)
+    {
+        this.gcpStorageBucket = requireNonNull(gcpStorageBucket, "gcpStorageBucket is null");
+
+        requireNonNull(gcpCredentialKey, "gcpCredentialKey is null");
+        InputStream jsonKey = new ByteArrayInputStream(Base64.getDecoder().decode(gcpCredentialKey));
+        try {
+            this.gcpCredentialsFile = Files.createTempFile("gcp-credentials", ".json", READ_ONLY_PERMISSIONS);
+            gcpCredentialsFile.toFile().deleteOnExit();
+            Files.write(gcpCredentialsFile, jsonKey.readAllBytes());
+
+            HiveGcsConfig gcsConfig = new HiveGcsConfig().setJsonKeyFilePath(gcpCredentialsFile.toAbsolutePath().toString());
+            Configuration configuration = ConfigurationInstantiator.newEmptyConfiguration();
+            new GoogleGcsConfigurationInitializer(gcsConfig).initializeConfiguration(configuration);
+
+            this.fileSystem = FileSystem.newInstance(new URI(bucketUrl()), configuration);
+        }
+        catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void removeTestData()
+    {
+        if (fileSystem != null) {
+            try {
+                fileSystem.delete(new org.apache.hadoop.fs.Path(bucketUrl()), true);
+            }
+            catch (IOException e) {
+                e.printStackTrace();
```

Comment:
I think that this is a leftover.
Please adjust the exception handling.

Reply:
That was on purpose actually. The bucket the CI is using has a three day expiration policy anyway, so it's not a big deal if the cleanup fails.

Reply:
use `log.warn` (or info) plus a code comment.

(`e.printStackTrace();` looks like auto-generated by an IDE, left over)